mailinglist/src/main/java/org/openjdk/skara/mailinglist/Mbox.java

```diff
@@ -89,45 +89,72 @@ private static String decodeFromStrings(String body) {
                                        .filter(email -> !email.hasHeader("In-Reply-To"))
                                        .collect(Collectors.toMap(Email::id, Conversation::new));
 
-        var outOfOrder = new ArrayList<Email>();
-        var lastOutOfOrderCount = -1;
-        while (outOfOrder.size() != lastOutOfOrderCount) {
-            lastOutOfOrderCount = outOfOrder.size();
-            outOfOrder.clear();
-
-            for (var email : emails) {
-                if (email.hasHeader("In-Reply-To")) {
-                    var inReplyToMatcher = inReplyToPattern.matcher(email.headerValue("In-Reply-To"));
-                    if (!inReplyToMatcher.find()) {
-                        log.info("Cannot parse In-Reply-To header: " + email.headerValue("In-Reply-To"));
-                        continue;
-                    }
-                    var inReplyTo = EmailAddress.from(inReplyToMatcher.group(1));
-                    if (!idToMail.containsKey(inReplyTo)) {
-                        log.info("Can't find parent: " + inReplyTo + " - discarding");
-                        continue;
-                    }
-                    var parent = idToMail.get(inReplyTo);
-                    if (!idToConversation.containsKey(inReplyTo)) {
-                        outOfOrder.add(email);
-                    } else {
-                        var conversation = idToConversation.get(inReplyTo);
-                        conversation.addReply(parent, email);
-                        idToConversation.put(email.id(), conversation);
+        var unhandledEmails = emails;
+        var lastUnhandledCount = 0;
+        while (unhandledEmails.size() != lastUnhandledCount) {
+            lastUnhandledCount = unhandledEmails.size();
+            var emailsToCheck = unhandledEmails;
+            unhandledEmails = new ArrayList<>();
+
+            for (var email : emailsToCheck) {
+                if (email.id().toString().contains("<VtLlAA6xMBK-6Ib-4SbkYZJssOuyqPl5Dr7kdiJUf6A=.9c549bc6-867f-4ba6-8727-efafe7cc18e2@github.com>")) {
+                    log.warning("Found first email of jdk#5300 in mbox");
```

Comment:
What is the purpose of this check?

Reply:
Ah, that's a left over debug message (and break point line) from my testing. Will remove.